File: leetcode/easy/add-strings/add-strings.ts
function addStrings(num1: string, num2: string): string {
  let longer: string, shorter: string;
  if (num2.length > num1.length) {
      longer = num2;
      shorter = num1;
  }
  else {
      longer = num1;
      shorter = num2;
  }

  for (let i = 0; i < longer.length - shorter.length; i++) {
      shorter = '0' + shorter;
  }

  let result = '';
  let carryOver: 0 | 1 = 0;

  for (let i = longer.length - 1; i >= 0; i--) {
    let digitSum: number = Number(longer.charAt(i)) + Number(shorter.charAt(i));
    if (digitSum >= 10) {
      digitSum -= 10;
      carryOver = 1;
    }  
    result += (digitSum + carryOver).toString();
    carryOver = 0;
  }

  return result;
};

File: leetcode/easy/add-strings/add-strings.java
class Solution {
    public String[] equalizeStrings(String num1, String num2) {
        if (num1.length() > num2.length()) {
            StringBuilder n2 = new StringBuilder(num2);
            while (n2.length() < num1.length()) {
                System.out.println(n2.length() + ", " + num1.length());
                n2.insert(0, '0');
            }
            return new String[] {num1, n2.toString()};

        }
        else if (num2.length() > num1.length()) {
            StringBuilder n1 = new StringBuilder(num1);
            while (n1.length() < num2.length()) {
                n1.insert(0, '0');
            }
            return new String[] {n1.toString(), num2};

        }
        return new String[] {num1, num2};
    }

    public String addStrings(String num1, String num2) {
        String[] adjustedStrings = equalizeStrings(num1, num2);
        num1 = adjustedStrings[0]; num2 = adjustedStrings[1];

        StringBuilder result = new StringBuilder();


        int i = num1.length() - 1;
        int carryOver = 0;
        
        while (i >= 0) {
            int d1 = Character.getNumericValue(num1.charAt(i));
            int d2 = Character.getNumericValue(num2.charAt(i));
            int digitSum = d1 + d2 + carryOver;
            if (digitSum >= 10) {
                carryOver = 1;
                digitSum -= 10;
            }
            else {
                carryOver = 0;
            }
            result.insert(0, (char) (digitSum + '0'));
            i--;
        }

        if (carryOver == 1) result.insert(0, '1');

        return result.toString();
    }
}